### PR TITLE
Document default values for primitive types

### PR DESCRIPTION
**Pull request by vbrandl**
All primitive types implement the `Default` trait but the documentation just says `Returns the "default value" for a type.` and doesn't give a hint about the actual default value. I think it would be good to document the default values in a proper way.
I changed the `default_impl` macro to accept a doc string as a third parameter and use this string to overwrite the documentation of `default()` for each primitive type.
The generated documentation now looks like this:
Documentation of default() on the bool primitive

### CODE REVIEW

**Comment by retep998** [comment]
Bikeshedding alternative phrasing:
`Returns the default value of 0`

**Comment by vbrandl** [comment]
Shouldn't @rust-highfive assign a reviewer to this PR? Or do I have to do it manually or just wait?

File: src/libcore/default.rs [hunk 1]
```diff
@@ -126,32 +126,33 @@ pub trait Default: Sized {
...
         }
     }
 }
 
-default_impl! { (), () }
-default_impl! { bool, false }
-default_impl! { char, '\x00' }
+default_impl! { (), (), "Returns the default value of `()`" }
+default_impl! { bool, false, "Returns the default value of `false`" }
+default_impl! { char, '\x00', "Returns the default value of `\\x00`" }
```
Comment (GuillaumeGomez) [inline src/libcore/default.rs]: Shouldn't it be "Returns the default value of `char` which is `\\x00`?"

File: src/libcore/default.rs [hunk 2]
```diff
@@ -126,32 +126,33 @@ pub trait Default: Sized {
...
             fn default() -> $t { $v }
         }
     }
 }
 
-default_impl! { (), () }
-default_impl! { bool, false }
-default_impl! { char, '\x00' }
+default_impl! { (), (), "Returns the default value of `()`" }
+default_impl! { bool, false, "Returns the default value of `false`" }
```
Comment (GuillaumeGomez) [inline src/libcore/default.rs]: Shouldn't it be "Returns the default value of`bool` which is `false`?"

**Review by GuillaumeGomez — CHANGES_REQUESTED** [review]
A light rewording would be better I think.

**Comment by GuillaumeGomez** [comment]
Hum, actually I think it's good as is. The wording is lighter. I'll go for this one then. Thanks!

@bors: r+ rollup

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**